sdks/java/io/kafka/src/main/java/org/apache/beam/sdk/io/kafka/KafkaReadSchemaTransformProvider.java

```diff
@@ -138,59 +158,23 @@ public PCollectionRowTuple expand(PCollectionRowTuple input) {
           PCollection<byte[]> kafkaValues =
               input.getPipeline().apply(kafkaRead.withoutMetadata()).apply(Values.create());
 
+          Schema errorSchema = ErrorHandling.errorSchema(ERROR_SCHEMA);
```

Comment:
This probably shouldn't be an ErrorSchema of ERROR_SCHEMA. 

We should add a errorSchema(FieldType) overload, where bytes is the thing to pass here. I know John Casey is looking at DLQ stuff where we might want to refactor this a bit more. (I suppose we're calling YAML stable, but the formatting of the error messages may have to change.)

Reply:
Ok, I did that initially, but I wan't sure if that would mess with our Row-based approach. I can refactor that and test

Reply:
https://github.com/apache/beam/pull/29274 (as this is in the docs, no need to cherry-pick that one).